mongoose.c

```diff
@@ -9308,12 +9308,12 @@ void ETH_IRQHandler(void) {
       if (++s_rxno >= ETH_DESC_CNT) s_rxno = 0;
     }
   }
-#ifdef __riscv
-  ETH->DMASR = ~0;   // TODO: do more fine-grained flag cleanup
-#else
-  ETH->DMASR = MG_BIT(7);  // Clear possible RBUS while processing
-#endif
-  ETH->DMARPDR = 0;  // and resume RX
+  // Cleanup flags
+  ETH->DMASR = (7UL < 20)    // TPS - Transmit process status
+               | (7UL < 17)  // RPS - Receive process status
```

Comment:
"These bits indicate the Receive DMA FSM state. This field does not generate an interrupt."
The case where this may stop (a burst arrives and no more descriptors are left) is handled at the end of the handler with `ETH->DMARPDR = 0`. This should not be necessary; furthermore, these bits are read-only, so this has no effect.